### PR TITLE
Simplify scala.runtime

### PR DESCRIPTION
**Pull request by sjrd**
This removes as much as I could in one day from the `scala.runtime` package.
Since everything in there, _especially_ the content of `ScalaRunTime`, is codegen-only API, it is supposed to be safe to remove things without deprecation periods.

### CODE REVIEW

**Comment by sjrd** [comment]
Tentatively assigning the review to @SethTisue, because you're listed as "library" in the readme.

**Comment by lrytz** [comment]
Hey! Since you've been looking at 36ef60e, maybe you can answer my question here: https://github.com/scala/scala/pull/5096/commits/16ed2ace86a79dfa6c3a43e58837d99c346f2f84#diff-2e0fd5042aa83407a80a7557cc2a666bR670

**Comment by lrytz** [comment]
LGTM, very nice!!!

File: test/junit/scala/runtime/ScalaRunTimeTest.scala [hunk 1]
```diff
@@ -5,70 +5,10 @@ import org.junit.Test
...
-    check((0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0))
-    check((0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0))
-
-    case class C()
-    val c = new C()
-    assertFalse(c.toString, isTuple(c))
-  }
-
-  @Test
   def testStingOf() {
```
Comment (som-snytt) [inline test/junit/scala/runtime/ScalaRunTimeTest.scala:12]: Death, where is thy sting?

Comment (sjrd) [inline test/junit/scala/runtime/ScalaRunTimeTest.scala:12]: I guess this is a call for me to fix that typo, as a boy scout?

Comment (som-snytt) [inline test/junit/scala/runtime/ScalaRunTimeTest.scala:12]: Only if you feel the call. I couldn't pass up a unit test pun. Are there French scouts? or only Legionnaires? My daughter just researched the origins of Girl Scouts as a cosmopolitan effort. I was just moved to go put on my "je suis scala.js" t-shirt.

**Pull request merged**